### PR TITLE
[REVIEW] Fix series dict alignment to not drop index name

### LINKED ISSUE #4211: [BUG] Dask cuDF MultiColumn groupby agg reset_index fails with non-matching metadata
Calling reset_index on Dask dataframe fails with index misalignment depending on the data distribution. See example below (conda nightly, 2020-02-20). This fails for me about 1/3 of the time. With more rows, the failure rate increases (until its functionally 100%).

```python
import cudf
import dask_cudf

df = cudf.datasets.randomdata(
    nrows=160, dtypes={"a":int, "c":int}
)
print(len(df.a.unique()))

ddf = dask_cudf.from_cudf(df, 1)
len(ddf.groupby(["a"]).agg({"c":["count", "mean"]}).reset_index())
90
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-37-c2e3b6370de4> in <module>
      8 
      9 ddf = dask_cudf.from_cudf(df, 1)
---> 10 len(ddf.groupby(["a"]).agg({"c":["count", "mean"]}).reset_index())

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/dataframe/core.py in __len__(self)
   3318             return super().__len__()
   3319         else:
-> 3320             return len(s)
   3321 
   3322     @property

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/dataframe/core.py in __len__(self)
    525     def __len__(self):
    526         return self.reduction(
--> 527             len, np.sum, token="len", meta=int, split_every=False
    528         ).compute()
    529 

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/base.py in compute(self, **kwargs)
    164         dask.base.compute
    165         """
--> 166         (result,) = compute(self, traverse=False, **kwargs)
    167         return result
    168 

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/base.py in compute(*args, **kwargs)
    435     keys = [x.__dask_keys__() for x in collections]
    436     postcomputes = [x.__dask_postcompute__() for x in collections]
--> 437     results = schedule(dsk, keys, **kwargs)
    438     return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
    439 

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/threaded.py in get(dsk, result, cache, num_workers, pool, **kwargs)
     82         get_id=_thread_get_id,
     83         pack_exception=pack_exception,
---> 84         **kwargs
     85     )
     86 

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/local.py in get_async(apply_async, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, **kwargs)
    484                         _execute_task(task, data)  # Re-execute locally
    485                     else:
--> 486                         raise_exception(exc, tb)
    487                 res, worker_id = loads(res_info)
    488                 state["cache"][key] = res

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/local.py in reraise(exc, tb)
    314     if exc.__traceback__ is not tb:
    315         raise exc.with_traceback(tb)
--> 316     raise exc
    317 
    318 

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/local.py in execute_task(key, task_info, dumps, loads, get_id, pack_exception)
    220     try:
    221         task, data = loads(task_info)
--> 222         result = _execute_task(task, data)
    223         id = get_id()
    224         result = dumps((result, id))

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]
--> 118         args2 = [_execute_task(a, cache) for a in args]
    119         return func(*args2)
    120     elif not ishashable(arg):

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in <listcomp>(.0)
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]
--> 118         args2 = [_execute_task(a, cache) for a in args]
    119         return func(*args2)
    120     elif not ishashable(arg):

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    113     """
    114     if isinstance(arg, list):
--> 115         return [_execute_task(a, cache) for a in arg]
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in <listcomp>(.0)
    113     """
    114     if isinstance(arg, list):
--> 115         return [_execute_task(a, cache) for a in arg]
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]
--> 118         args2 = [_execute_task(a, cache) for a in args]
    119         return func(*args2)
    120     elif not ishashable(arg):

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in <listcomp>(.0)
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]
--> 118         args2 = [_execute_task(a, cache) for a in args]
    119         return func(*args2)
    120     elif not ishashable(arg):

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    113     """
    114     if isinstance(arg, list):
--> 115         return [_execute_task(a, cache) for a in arg]
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in <listcomp>(.0)
    113     """
    114     if isinstance(arg, list):
--> 115         return [_execute_task(a, cache) for a in arg]
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]
--> 118         args2 = [_execute_task(a, cache) for a in args]
    119         return func(*args2)
    120     elif not ishashable(arg):

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in <listcomp>(.0)
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]
--> 118         args2 = [_execute_task(a, cache) for a in args]
    119         return func(*args2)
    120     elif not ishashable(arg):

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    113     """
    114     if isinstance(arg, list):
--> 115         return [_execute_task(a, cache) for a in arg]
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in <listcomp>(.0)
    113     """
    114     if isinstance(arg, list):
--> 115         return [_execute_task(a, cache) for a in arg]
    116     elif istask(arg):
    117         func, args = arg[0], arg[1:]

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    117         func, args = arg[0], arg[1:]
    118         args2 = [_execute_task(a, cache) for a in args]
--> 119         return func(*args2)
    120     elif not ishashable(arg):
    121         return arg

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/optimization.py in __call__(self, *args)
    980         if not len(args) == len(self.inkeys):
    981             raise ValueError("Expected %d args, got %d" % (len(self.inkeys), len(args)))
--> 982         return core.get(self.dsk, self.outkey, dict(zip(self.inkeys, args)))
    983 
    984     def __reduce__(self):

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in get(dsk, out, cache)
    147     for key in toposort(dsk):
    148         task = dsk[key]
--> 149         result = _execute_task(task, cache)
    150         cache[key] = result
    151     result = _execute_task(out, cache)

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/core.py in _execute_task(arg, cache, dsk)
    117         func, args = arg[0], arg[1:]
    118         args2 = [_execute_task(a, cache) for a in args]
--> 119         return func(*args2)
    120     elif not ishashable(arg):
    121         return arg

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/utils.py in apply(func, args, kwargs)
     27 def apply(func, args, kwargs=None):
     28     if kwargs:
---> 29         return func(*args, **kwargs)
     30     else:
     31         return func(*args)

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/dataframe/core.py in apply_and_enforce(*args, **kwargs)
   5050             return meta
   5051         if is_dataframe_like(df):
-> 5052             check_matching_columns(meta, df)
   5053             c = meta.columns
   5054         else:

~/miniconda3/envs/rapids-20200220-cuda101/lib/python3.7/site-packages/dask/dataframe/utils.py in check_matching_columns(meta, actual)
    676             " the columns in the provided metadata\n"
    677             "  Extra:   %s\n"
--> 678             "  Missing: %s" % (extra, missing)
    679         )
    680 

ValueError: The columns in the computed data do not match the columns in the provided metadata
  Extra:   [('index', '')]
  Missing: [('a', '')]
```

### PR DESCRIPTION
**Pull request by kkraus14**
Fixes #4211

### CODE REVIEW

File: python/cudf/cudf/core/index.py [hunk 1]
```diff
@@ -97,7 +97,20 @@ def deserialize(cls, header, frames):
 
     @property
     def names(self):
-        return (self.name,)
+        return [self.name]
```
Comment (jakirkham) [inline python/cudf/cudf/core/index.py]: Should this still be a `tuple` in the rare event someone tries to assign to this later (without the setter being involved)? Would help catch any errors in that event 🙂

Comment (kkraus14) [reply python/cudf/cudf/core/index.py]: This actually can't be a tuple as trying to set the names of an index via a tuple is not allowed to protect against the case where the name is actually a tuple.

Comment (kkraus14) [reply python/cudf/cudf/core/index.py]: Pandas uses their own `FrozenList` for this reason but it felt weird to reach deep inside of Pandas to use that here.

Comment (kkraus14) [reply python/cudf/cudf/core/index.py]: I was actually incorrect here and misread an exception from passing a string directly as opposed to a tuple 😄

**Pull request merged**